spm_functions.py

```diff
@@ -1,3 +1,102 @@
 # spm_functions.py
 #
-#  This file holds utility functions called by the spm model.
\ No newline at end of file
+#  This file holds utility functions called by the spm model.
+
+import numpy as np 
+import math
+
+def Half_Cell_Eqlib_Potential(HalfCell,F = 96.48534, T_amb = 298.15, R = 0.0083145):
+    """
+    Help for this function.
+    """
+    # F = 96.48534 #Faraday's number [kC/equivalence]
+    # R = 0.0083145 #Universal gas constant [kJ/mol-K]
+    n_elc = HalfCell.n
+    print("n",n_elc)
+    #T_amb = 273.15 + 25 #[K]
+    T = HalfCell.Temp
+     
+    Delta_G_cell = np.dot(HalfCell.G,HalfCell.nu)
+    Delta_S = np.dot(HalfCell.S,HalfCell.nu)
+
+    U_0_Cell_amb =  -Delta_G_cell/(n_elc*F)
+    U_0_Cell = U_0_Cell_amb + (T- T_amb)*Delta_S/(n_elc*F)
+    U_Cell = U_0_Cell - R*T/n_elc/F*np.log(np.prod(np.power(HalfCell.X,HalfCell.nu)))
+    return U_Cell
+
+def current_density(i_o,V,U,T,F = 96.48534, Beta = 0.5, R = 0.0083145, n = 1):
+    """
+    The number this returns is absolutly massive, but I cannot find an error in my equation
+    
+    Parameters
+    ----------
+    i_o : Exchange Current Density [mA/cm^2] 
+    V : Electrode potential difference at the electrode-electrolyte interface [V]
```

Comment:
```suggestion
    V : Electric potential difference at the electrode-electrolyte interface (phi_ed - phi_elyte) [V]
```

 Check that I understand your sign convention correctly.